Subject: [PATCH] Fix food confirmation
---
Index: app/(content)/bookings/page.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/(content)/bookings/page.tsx b/app/(content)/bookings/page.tsx
--- a/app/(content)/bookings/page.tsx	(revision 2370cc08cd255b0567a0b78239d527b69e00b911)
+++ b/app/(content)/bookings/page.tsx	(revision 0407e76d28f3b207f47d67082c842e67830a16af)
@@ -136,24 +136,27 @@
                     orderRef.current?.showModal();
             }}>{payment?.transactionStatus === 'SUCCESS' ? 'Вы уже оплатили заказ' : 'Добавить еды'}
             </button>
-            {orders.length > 0 && orderDetails > -1 && <button disabled={payment?.transactionStatus === 'SUCCESS'} onClick={() => {
-                if (changed) {
-                    ConfirmOrder(orderDetails)
-                        .then(orders => {
-                            if (orders) setOrders(orders.orders);
-                            if (orders) setChanged(false);
-                        });
-                } else {
-                    if (payment)
-                        payRef.current?.showModal();
-                    else
-                        InitPayment(orderDetails).then(pay => {
-                            setPayment(pay);
-                            payRef.current?.showModal();
-                        })
-                }
-            }}>{changed ? 'Подтвердить еду' : payment?.transactionStatus === 'SUCCESS' ? 'Оплачено' : 'Оплатить'}
-            </button>}
+            {orders.length > 0 && orderDetails > -1 &&
+                <button disabled={payment?.transactionStatus === 'SUCCESS'} onClick={() => {
+                    if (changed) {
+                        ConfirmOrder(orderDetails)
+                            .then(orders => {
+                                if (orders) {
+                                    setOrders(orders.orders);
+                                    setTimeout(() => setChanged(false), 100);
+                                }
+                            });
+                    } else {
+                        if (payment)
+                            payRef.current?.showModal();
+                        else
+                            InitPayment(orderDetails).then(pay => {
+                                setPayment(pay);
+                                payRef.current?.showModal();
+                            })
+                    }
+                }}>{changed ? 'Подтвердить еду' : payment?.transactionStatus === 'SUCCESS' ? 'Оплачено' : 'Оплатить'}
+                </button>}
             <button onClick={deleteBooking}>Отменить бронь
             </button>
             {payment && <PaymentPopup ref={payRef} payment={payment} setPayment={paym => {
